### PR TITLE
Version 1.6.2

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # Change log for kotlinx.coroutines
 
+## Version 1.6.2
+
+* Fixed a bug with `ThreadLocalElement` not being correctly updated when the most outer `suspend` function was started directly without `kotlinx.coroutines` (#2930).
```
Comment (dkhalanskyjb) [inline CHANGES.md:5]:
```suggestion
* Fixed a bug with `ThreadLocalElement` not being correctly updated when the outermost `suspend` function was called directly without `kotlinx.coroutines` (#2930).
```

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+* Fixed multiple data races that might have been affecting `Mutex` and `runBlocking` event loop (#3250, #3251).
```
Comment (dkhalanskyjb) [inline CHANGES.md:6]: The data race in `Mutex` was benign (that is, shouldn't affect anything in practice), maybe we should highlight it.

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+* Obsolete `TestCoroutineContext` is removed along with JPMS split-package between `kotlinx-coroutines-core` and `kotlinx-coroutines-test` (#3218).
```
Comment (dkhalanskyjb) [inline CHANGES.md:7]:
```suggestion
* Obsolete `TestCoroutineContext` is removed, which fixes the `kotlinx-coroutines-test` JPMS package being split between `kotlinx-coroutines-core` and `kotlinx-coroutines-test` (#3218).
```

File: CHANGES.md [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+* Updated ProGuard rules that help further shrink the size of the resulting DEX file with coroutines (#3263).
```
Comment (dkhalanskyjb) [inline CHANGES.md:8]:
```suggestion
* Updated the ProGuard rules to further shrink the size of the resulting DEX file with coroutines (#3263).
```

Comment (dkhalanskyjb) [reply CHANGES.md:8]: Also, attribution.

File: CHANGES.md (hunk 2 above)
Comment (qwwdfsad) [reply CHANGES.md:6]: I'm not sure it really matters here

Comment (dkhalanskyjb) [reply CHANGES.md:6]: I just think that reading the changelog and seeing that `Mutex` had a data race is pretty scary, given that its whole purpose is to guard against them.

File: CHANGES.md [hunk 5]
```diff
@@ -1,5 +1,15 @@
 # Change log for kotlinx.coroutines
 
+## Version 1.6.2
+
+* Fixed a bug with `ThreadLocalElement` not being correctly updated when the most outer `suspend` function was called directly without `kotlinx.coroutines` (#2930).
+* Fixed multiple data races that might have been affecting `Mutex` and `runBlocking` event loop (#3250, #3251).
+* Obsolete `TestCoroutineContext` is removed, which fixes the `kotlinx-coroutines-test` JPMS package being split between `kotlinx-coroutines-core` and `kotlinx-coroutines-test` (#3218).
+* Updated the ProGuard rules to further shrink the size of the resulting DEX file with coroutines (#3263). Thanks @agrieve!
```
Comment (dkhalanskyjb) [inline CHANGES.md:8]:
```suggestion
* Updated the ProGuard rules to further shrink the size of the resulting DEX file with coroutines (#3111). Thanks, @agrieve!
```
We usually reference the issues, right?

File: CHANGES.md [hunk 6]
```diff
@@ -1,5 +1,15 @@
...
+* Atomicfu is updated to `0.17.2`, which includes a more efficient and robust JS IR transformer (#3255).
+* Kotlin is updated to `1.6.21`, Gradle version is updated to `7.4.2` (#3281). Thanks @wojtek-kalicinski!
```
Comment (dkhalanskyjb) [inline CHANGES.md:10]:
```suggestion
* Kotlin is updated to `1.6.21`, Gradle version is updated to `7.4.2` (#3281). Thanks, @wojtek-kalicinski!
```
If we're treating the changelog as a somewhat official document.

File: CHANGES.md (hunk 5 above)
Comment (qwwdfsad) [reply CHANGES.md:8]: I was thinking the PR is more insightful. I've left both references

File: CHANGES.md (hunk 6 above)
Comment (qwwdfsad) [reply CHANGES.md:10]: Done

File: CHANGES.md [hunk 7]
```diff
@@ -1,5 +1,15 @@
 # Change log for kotlinx.coroutines
 
+## Version 1.6.2
+
+* Fixed a bug with `ThreadLocalElement` not being correctly updated when the most outer `suspend` function was called directly without `kotlinx.coroutines` (#2930).
+* Fixed multiple data races that might have been affecting `runBlocking` event loop and benign data race in `Mutex` (#3250, #3251).
```
Comment (dkhalanskyjb) [inline CHANGES.md:6]:
```suggestion
* Fixed multiple data races: one that might have been affecting `runBlocking` event loop, and a benign data race in `Mutex` (#3250, #3251).
```

**Review by dkhalanskyjb — APPROVED** [review]

**Pull request merged**